bottlecap/src/proc/mod.rs

```diff
@@ -180,6 +211,135 @@ fn get_uptime_from_path(path: &str) -> Result<f64, io::Error> {
     ))
 }
 
+#[must_use]
+pub fn get_fd_max_data(pids: &[i32]) -> f64 {
+    get_fd_max_data_from_path(PROC_PATH, pids)
+}
+
+fn get_fd_max_data_from_path(path: &str, pids: &[i32]) -> f64 {
+    let mut fd_max: f64 = 1024.0; // Default to hard limit set by AWS Lambda
+
+    for &pid in pids {
+        let limits_path = format!("{path}/{pid}/limits");
+        let Ok(file) = File::open(&limits_path) else {
+            continue;
+        };
+
+        let reader = io::BufReader::new(file);
+        for line in reader.lines().map_while(Result::ok) {
+            if line.starts_with("Max open files") {
+                let values: Vec<&str> = line.split_whitespace().collect();
+
+                // Line should have 6 items: Max open files [soft limit] [hard limit] [units]
+                if values.len() < 6 {
+                    debug!("File descriptor max data not found in file {}", limits_path);
+                    break;
+                }
+
+                // Skip past the first three items ("Max open files") to get the soft limit
+                let fd_max_pid_str = values[3];
```

Comment:
Regexp are usually a good choice when parsing strings like this.
They are very efficient, requires less code and they are more robust than manual parsing. Chat gpt does a great job generating regexp given a string example